src/components/selector_menu/index.js

```diff
@@ -128,23 +126,30 @@ var SelectorMenu = React.createClass({
       return;
     }
 
-    var visible = _.map(fuzzy.filter(filterBy, this.state.visible), function(result) {
-      return result.string;
-    });
+    var visible = _.pluck(fuzzy.filter(filterBy, this.getOptionNames()), 'string');
 
     this.setState({
       visible: visible
     });
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    if (nextProps.optionsList.length !== this.props.optionsList.length) {
+      this.setState({ visible: [] });
+    }
+  },
+
   render: function() {
     var wrapperClass = this.state.expanded ? 'selector__wrapper expanded' : 'selector__wrapper';
     var innerClass = this.state.expanded ? 'inner-wrapper expanded' : 'inner-wrapper';
+    var visible = this.state.visible.length > 0 ? this.state.visible : this.getOptionNames();
+    var selected = this.state.selected && this.props.defaultSelection === undefined ?
```

Comment:
So ideally I'd like this to behave like a controlled component... Because the problem with having `this.state.selected` always control the value is that when an external change happens (like in the filters UI, where the filters are controlled in 2 different places) it gets stuck in the wrong state. 

Maybe the correct thing to do is follow the convention of react form inputs and have separate `defaultValue` and `value` props. In our case, `defaultSelection` would behave the same, supplying an initial value that's then entirely controlled with internally-tracked state (making it an "uncontrolled" component).

We would also expose a `value` (or `selection` to match the convention of the other name) that would take precedence over any internal state, so this can be used as a "controlled" component.

@florapdx lmk if you want to hangout on this today